### PR TITLE
Fixes processing 429s from SPs

### PR DESCRIPTION
**Pull request by brendankowitz**
## Description
Stored procedures return a statuscode 400 with a substatus when they fail. When this was a 429, we were not processing this correctly.
The errorcode in the SP was also not correct.

## Related issues
Addresses [issue #].

## Testing
Describe how this change was tested.

## FHIR Team Checklist
- [ ] **Update the title** of the PR to be succinct and less than 50 characters
- [ ] **Add a milestone** to the PR for the sprint that it is merged (i.e. add S47)
- [ ] Tag the PR with the type of update: **Bug**, **Dependencies**, **Enhancement**, or **New-Feature**
- [ ] Tag the PR with **Azure API for FHIR** if this will release to the managed service
- Review [squash-merge requirements](https://github.com/microsoft/fhir-server/blob/master/SquashMergeRequirements.md)

### Semver Change ([docs](https://github.com/microsoft/fhir-server/blob/master/docs/Versioning.md))
Patch|Skip|Feature|Breaking (reason)

### CODE REVIEW

**Review by rotodd — APPROVED** [review]

File: src/Microsoft.Health.Fhir.CosmosDb/Features/Storage/StoredProcedures/AcquireExportJobs/acquireExportJobs.js [hunk 1]
```diff
@@ -145,6 +145,6 @@ function acquireExportJobs(maximumNumberOfConcurrentJobsAllowedInString, jobHear
     }
 
     function throwTooManyRequestsError() {
-        throw new Error(ErrorCodes.RequestEntityTooLarge, `The request could not be completed.`);
+        throw new Error(429, `The request could not be completed.`);
```
Comment (Ivanidzo4ka) [inline src/Microsoft.Health.Fhir.CosmosDb/Features/Storage/StoredProcedures/AcquireExportJobs/acquireExportJobs.js:148]: >429 [](http://example.com/codeflow?start=24&length=3)

ErrorCodes.TooManyRequests sound more appropriate than a const

File: src/Microsoft.Health.Fhir.CosmosDb/Features/Storage/StoredProcedures/HardDelete/hardDelete.js [hunk 2]
```diff
@@ -24,6 +24,7 @@ function hardDelete(resourceTypeName, resourceId, keepCurrentVersion) {
 
     let deletedResourceIdList = new Array();
 
+    throwTooManyRequestsError();
```
Comment (Ivanidzo4ka) [inline src/Microsoft.Health.Fhir.CosmosDb/Features/Storage/StoredProcedures/HardDelete/hardDelete.js:27]: >throwTooManyRequestsError [](http://example.com/codeflow?start=4&length=25)

I don't get this addition

Comment (brendankowitz) [reply src/Microsoft.Health.Fhir.CosmosDb/Features/Storage/StoredProcedures/HardDelete/hardDelete.js:27]: Thanks, that was my manual test :(

File: src/Microsoft.Health.Fhir.CosmosDb/Features/Storage/StoredProcedures/AcquireExportJobs/acquireExportJobs.js (hunk 1 above)
Comment (brendankowitz) [reply src/Microsoft.Health.Fhir.CosmosDb/Features/Storage/StoredProcedures/AcquireExportJobs/acquireExportJobs.js:148]: This const isn't in provided out of the box :( http://azure.github.io/azure-cosmosdb-js-server/DocDbWrapperScript.js.html

I could create our own, but it is still defined per proc

**Pull request merged**